sys/net/link_layer/ng_nomac/ng_nomac.c

```diff
@@ -0,0 +1,155 @@
+    msg_init_queue(msg_queue, NG_NOMAC_MSG_QUEUE_SIZE);
+    /* save the PID to the device descriptor and register the device */
+    dev->mac_pid = thread_getpid();
+    ng_netif_add(dev->mac_pid);
+    /* register the event callback with the device driver */
+    dev->driver->add_event_callback(dev, _event_cb);
+
+    /* start the event loop */
+    while (1) {
+        DEBUG("nomac: waiting for incoming messages\n");
+        msg_receive(&msg);
+        /* dispatch NETDEV and NETAPI messages */
+        switch (msg.type) {
+            case NG_NETDEV_MSG_TYPE_EVENT:
+                DEBUG("nomac: NG_NETDEV_MSG_TYPE_EVENT received\n");
+                dev->driver->isr_event(dev, msg.content.value);
+                break;
+            case NG_NETAPI_MSG_TYPE_SND:
+                DEBUG("nomac: NG_NETAPI_MSG_TYPE_SND received\n");
+                dev->driver->send_data(dev, (ng_pktsnip_t *)msg.content.ptr);
+                break;
+            case NG_NETAPI_MSG_TYPE_SET:
+                /* TODO: filter out MAC layer options -> for now forward
+                         everything to the device driver */
+                DEBUG("nomac: NG_NETAPI_MSG_TYPE_SET received\n");
+                /* read incoming options */
+                opt = (ng_netapi_opt_t *)msg.content.ptr;
+                /* set option for device driver */
+                res = dev->driver->set(dev, opt->opt, opt->data, opt->data_len);
+                /* send reply to calling thread */
+                reply.type = NG_NETAPI_MSG_TYPE_ACK;
+                reply.content.value = (uint32_t)res;
+                msg_reply(&msg, &reply);
+                break;
+            case NG_NETAPI_MSG_TYPE_GET:
+                /* TODO: filter out MAC layer options -> for now forward
+                         everything to the device driver */
+                DEBUG("nomac: NG_NETAPI_MSG_TYPE_GET received\n");
+                /* read incoming options */
+                opt = (ng_netapi_opt_t *)msg.content.ptr;
+                /* get option from device driver */
+                res = dev->driver->get(dev, opt->opt, opt->data,
+                                       (size_t*)(&(opt->data_len)));
+                /* send reply to calling thread */
+                reply.type = NG_NETAPI_MSG_TYPE_ACK;
+                reply.content.value = (uint32_t)res;
+                msg_reply(&msg, &reply);
+                break;
+        }
+    }
+    /* never reached */
+    return NULL;
+}
+
+kernel_pid_t ng_nomac_init(char *stack, int stacksize, char priority,
+                        const char *name, ng_netdev_t *dev)
+{
+    kernel_pid_t res;
+
+    /* check if given netdev device is defined */
+    if (dev == NULL) {
+        return -ENODEV;
+    }
+    /* create new NOMAC thread */
+    res = thread_create(stack, stacksize, priority, CREATE_STACKTEST,
+                         _nomac_thread, (void *)dev, name);
+    if (res <= 0) {
+        return -EINVAL;
```

Comment:
I return `-EINVAL` explicitly here to comply with the documentation in `ng_nomac.h`...